src/UIAutomation/UiaOperationAbstraction/UiaOperationAbstraction.h

```diff
@@ -123,6 +137,50 @@ namespace UiaOperationAbstraction
             }
         }
 
+        template<class TryBody, class CatchBody>
+        void Try(TryBody&& tryBody, CatchBody&& catchBody) const
+        {
+            if (m_useRemoteApi)
+            {
+                auto newCatchBody = [this, catchBody(std::forward<CatchBody>(catchBody))]()
+                {
+                    catchBody(UiaFailure(m_remoteOperation, m_useRemoteApi));
+                };
+
+                m_remoteOperation.TryBlock(std::forward<TryBody>(tryBody), std::move(newCatchBody));
+            }
+            else
+            {
+                try
+                {
+                    tryBody();
+                }
+                catch(...)
+                {
+                    catchBody(UiaFailure(m_remoteOperation, m_useRemoteApi));
+                }
+            }
+        }
+
+        template<class TryBody>
+        void Try(TryBody&& tryBody) const

```

Comment:
We could probably simplify this by just calling `Try(std::forward<TryBody>(tryBody), [](UiaFailure /*failure*/) {});` internally.

Reply:
```cpp
template<class TryBody>
void Try(TryBody&& tryBody) const
{
    Try(std::forward<TryBody>(tryBody), [](UiaFailure /*failure*/){});
}
```